### PR TITLE
New freezing point calculation in share code. 

### PR DESCRIPTION
**Pull request by dabail10**
This should be part of cesm1_5_alpha06d along with a CICE and POP tag.

### CODE REVIEW

File: components/data_comps/dice/dice_comp_mod.F90 [hunk 1]
```diff
@@ -6,6 +6,7 @@ module dice_comp_mod
 ! !USES:
 
   use shr_const_mod
+  use shr_frz_mod
```
Comment (jedwards4b) [inline components/data_comps/dice/dice_comp_mod.F90]: Please use the only clause and explicitly list what you are using.

File: components/data_comps/docn/docn_comp_mod.F90 [hunk 2]
```diff
@@ -6,6 +6,7 @@ module docn_comp_mod
 ! !USES:
 
   use shr_const_mod
+  use shr_frz_mod
```
Comment (jedwards4b) [inline components/data_comps/docn/docn_comp_mod.F90]: please add only clause

**Comment by jedwards4b** [comment]
Review complete, please add only clause to use statements - otherwise all looks good.

**Comment by dabail10** [comment]
Only clauses added.

**Comment by jtruesdal** [comment]
Hi:

I'll run aux_pop on the changes and will make the pop tag after that
finishes.

jt

**Pull request merged**